brainscore_vision/model_helpers/activations/core.py

```diff
@@ -231,6 +379,115 @@ def insert_attrs(self, wrapper):
+                               'extent of microsaccades.')
+
+        # compute the maximum radius of microsaccade extent in pixel space
+        if self._visual_degrees is None:
+            raise AssertionError('self._visual_degrees is not set by the ModelCommitment, but microsaccades '
+                                 'are in use. Set activations_model visual degrees in your commitment after defining '
+                                 'your activations_model. For example, self.activations_model.set_visual_degrees'
+                                 '(visual_degrees). For detailed information, see '
+                                 ':meth:`~brainscore_vision.model_helpers.activations.ActivationsExtractorHelper.'
+                                 '__call__`,')
+        radius_ratio = self.microsaccade_extent_degrees / self._visual_degrees
+        max_radius = radius_ratio * image_shape[0]  # maximum radius in pixels, set in self.microsaccade_extent_degrees
+
+        selected_microsaccades = {'pixels': [], 'degrees': []}
+        # microsaccades are placed in a spiral at sub-pixel increments
+        a = max_radius / np.sqrt(self.number_of_trials)  # spiral coefficient to space microsaccades evenly
+        for i in range(self.number_of_trials):
+            r = np.sqrt(i / self.number_of_trials) * max_radius  # compute radial distance for the i-th point
+            theta = a * np.sqrt(i) * 2 * np.pi / max_radius  # compute angle for the i-th point
+
+            # convert polar coordinates to Cartesian, centered on the image
+            x = r * np.cos(theta)
+            y = r * np.sin(theta)
+
+            pixels_per_degree = self.calculate_pixels_per_degree_in_image(image_shape[0])
+            selected_microsaccades['pixels'].append((x, y))
+            selected_microsaccades['degrees'].append(self.convert_pixels_to_degrees((x, y), pixels_per_degree))
+
+        # to keep consistent with number_of_trials, we count trial_number from 1 instead of from 0
+        self.microsaccades['pixels'][image_path] = selected_microsaccades['pixels']
+        self.microsaccades['degrees'][image_path] = selected_microsaccades['degrees']
+        return selected_microsaccades['pixels'][trial_number - 1]
+
+    def unpack_microsaccade_coords(self, stimuli_paths: np.ndarray, pixels_or_degrees: str,
+                            dim: int):
+        """Unpacks microsaccades from stimuli_paths into a single list to conform with coord requirements."""
+        # Python 3.7 does not support typing.Literal
+        assert pixels_or_degrees == 'pixels' or pixels_or_degrees == 'degrees'
+        unpacked_microsaccades = []
+        for stimulus_path in stimuli_paths:
+            for microsaccade in self.microsaccades[pixels_or_degrees][stimulus_path]:
+                unpacked_microsaccades.append(microsaccade[dim])
+        return unpacked_microsaccades
+
+    def calculate_pixels_per_degree_in_image(self, image_width_pixels: int) -> float:
```

Comment:
related to many methods, but it might be good to group all the micro-saccade stuff into their own class, e.g. `MicrosaccadeHelper` or so